### ISSUE TITLE
display-name enhancement: disallow specified names

### ISSUE THREAD

**Issue by peet** [yannickcr/eslint-plugin-react#650]
The problem that I'm having, which I'd like to enforce against, is code like this:

```js
module.exports = React.createClass({
});
```

which gets a name of 'exports' and is entirely unhelpful in the React dev tools and console messages.

Either assigning to a variable first, or setting the `displayName` prop explicitly would be fine.

My proposal is to add config to the 'display-name` rule to disallow certain names, e.g
```json
"display-name": [2, { "forbid": ["exports", "default"] }]
```

My feeling is that this should maybe only apply to inferred/transpiler names; if you want to explicitly name your component 'exports' with 'displayName', then that should be allowed. Might be tough to implement that distinction, however.

**Comment by yannickcr** [comment]
The `react/display-name` rule will already prevent this case, it will accept a `displayName` property or a named export but will throw if you try to do an unnamed export like your example.
